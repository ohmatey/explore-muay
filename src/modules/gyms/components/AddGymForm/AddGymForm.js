import React from 'react'

import TextField from '../../../common/form/TextField'

const AddGymForm = ({
  handleSubmit
}) => (
  <form onSubmit={handleSubmit}>
    <TextField name='gymName' />

    <TextField name='gymOwner' />

    <button type='submit'>Submit</button>
  </form>
)

export default AddGymForm
