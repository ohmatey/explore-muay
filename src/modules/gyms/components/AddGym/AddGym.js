import React from 'react'
import PropTypes from 'prop-types'
import { Form } from 'react-final-form'

import AddGymForm from '../AddGymForm'

const AddGym = ({
  handleFormSubmit
}) => (
  <div>
    <Form
      onSubmit={handleFormSubmit}
      component={AddGymForm}
    />
  </div>
)

AddGym.propsTypes = {
  handleFormSubmit: PropTypes.func.isRequired
}

export default AddGym
