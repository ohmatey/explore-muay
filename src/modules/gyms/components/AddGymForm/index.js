import AddGymForm from './AddGymForm'

export default AddGymForm
