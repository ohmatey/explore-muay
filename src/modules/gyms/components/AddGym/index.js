import AddGym from './AddGym'

export default AddGym
