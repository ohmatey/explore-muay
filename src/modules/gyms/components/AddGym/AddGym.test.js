import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Form } from 'react-final-form'

import AddGym from './AddGym'
import AddGymForm from '../AddGymForm'

describe('<AddGym />', () => {
  let component

  beforeEach(() => {
    component = shallow(
      <AddGym
        handleFormSubmit={jest.fn()}
      />
    )
  })

  it('should render with no errors', () => {
    expect(toJson(component)).toMatchSnapshot()
  })

  it('should render Form', () => {
    const mockHandleSubmit = jest.fn()

    component.setProps({
      handleFormSubmit: mockHandleSubmit
    })

    const form = component.find(Form)

    expect(form.prop('onSubmit')).toEqual(mockHandleSubmit)
    expect(form.prop('component')).toEqual(AddGymForm)
  })
})
