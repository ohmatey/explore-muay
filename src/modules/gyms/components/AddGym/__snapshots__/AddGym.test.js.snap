// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddGym /> should render with no errors 1`] = `
<div>
  <ReactFinalForm
    component={[Function]}
    onSubmit={[MockFunction]}
  />
</div>
`;
